### ISSUE TITLE
Feature request: including filenames in dask bags

### ISSUE THREAD

**Issue by SultanOrazbayev** [dask/dask#5810]
Is it possible to introduce file names that are read using `dask.bag.read_text`?

My use case is that I read json files into a dask bag, but every now and then the underlying json contains an error/inconsistent format, so my pipeline breaks. Knowing the filenames would make it somewhat more convenient in locating these files.

I think it's only of minor importance, and I'm not even sure how this could be implemented (it's more straightforward for dataframes: https://github.com/dask/dask/issues/2802)... maybe for text files a line appended/prepended or for json files a key appended.

**Comment by TomAugspurger** [comment]
Do you have a proposal for where the filename would be inserted in the data structure?

IMO, this would be more cleanly done using `dask.delayed` to return a tuple of `(filename, data_structure)`. You could perhaps do something like

```python
bag = db.read_text("*.json")
# inspect bag.dask for the filenames discovered
bag.map_partitions(join_with_filenames)
```

**Comment by TomAugspurger** [comment]
Closed by #5836
